### PR TITLE
ROX-19094: Changed SUCCESS to DELIVERED and GENERATED

### PR DESCRIPTION
**Pull request by sachaudh**
## Description

This PR adds a change to the Report Status by removing the `SUCCESS` state and adding the `DELIVERED` and `GENERATED` states. 

NOTE: This PR should be merged at the same time as https://github.com/stackrox/stackrox/pull/7634 in order to make sure things don't break

Screenshot 2023-09-05 at 12 38 55 PM
Screenshot 2023-09-05 at 12 39 06 PM

### CODE REVIEW

File: ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobStatus.tsx [hunk 1]
```diff
@@ -104,10 +99,10 @@ function ReportJobStatus({
                 </FlexItem>
             </Flex>
         );
-    } else if (reportStatus.runState === 'SUCCESS') {
+    } else if (reportStatus.runState === 'DELIVERED') {
         statusColorClass = 'pf-u-success-color-100';
-        statusIcon = <CheckCircleIcon title="Report run was successful" />;
-        statusText = <p className="pf-u-success-color-100">Successful</p>;
+        statusIcon = <CheckCircleIcon title="Report was successfully delivered" />;
+        statusText = <p className="pf-u-success-color-100">Successfully delivered</p>;
```
Comment (sachaudh) [inline ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobStatus.tsx:105]: @zhenpesky I changed the text from `Successful` to `Successfully delivered` for when a report run successfully delivers a report to the user's email. Let me know if that's fine or if you still prefer what we had before. I think this is clearer

File: ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobStatus.tsx [hunk 2]
```diff
@@ -119,9 +114,17 @@ function ReportJobStatus({
...
         statusIcon = <PendingIcon title="Report run is waiting" />;
         statusText = <p>Waiting</p>;
+    } else {
+        statusColorClass = 'pf-u-warning-color-100';
+        statusIcon = (
+            <Tooltip content="Please contact support for more help.">
+                <ExclamationTriangleIcon title="Report run status is unavailable" />
+            </Tooltip>
+        );
+        statusText = <p>Unknown status</p>;
```
Comment (sachaudh) [inline ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobStatus.tsx:127]: @zhenpesky When a status doesn't match anything we have, I added this. I noticed when the status change was made on the backend, the UI was just displaying `WAITING`. So this makes it easier to tell something is wrong. Let me know if you have better wording here. Also, let me know if there is a better icon for this

File: ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobs.tsx [hunk 3]
```diff
@@ -109,7 +109,10 @@ function ReportJobs({ reportId }: RunHistoryProps) {
                         >
                             <SelectOption value={runStates.PREPARING}>Preparing</SelectOption>
                             <SelectOption value={runStates.WAITING}>Waiting</SelectOption>
-                            <SelectOption value={runStates.SUCCESS}>Successful</SelectOption>
+                            <SelectOption value={runStates.GENERATED}>
+                                Download generated
+                            </SelectOption>
+                            <SelectOption value={runStates.DELIVERED}>Email delivered</SelectOption>
```
Comment (sachaudh) [inline ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobs.tsx:115]: @zhenpesky Let me know if the filter options text looks good to you

Screenshot 2023-09-05 at 12 39 06 PM

File: ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobStatus.tsx (hunk 1 above)
Comment (zhenpesky) [reply ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobStatus.tsx:105]: sounds good

File: ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobStatus.tsx (hunk 2 above)
Comment (zhenpesky) [reply ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobStatus.tsx:127]: looks good

File: ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobs.tsx (hunk 3 above)
Comment (zhenpesky) [reply ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobs.tsx:115]: LGTM

**Review by vjwilson — APPROVED** [review]

**Review by dvail — APPROVED** [review]

**Pull request merged**